contracts/src/Assets.sol

```diff
@@ -174,4 +179,112 @@ library Assets {
 
         emit IGateway.TokenRegistrationSent(token);
     }
+
+    // @dev Transfer polkadot native tokens back
+    function sendForeignToken(
+        address agent,
+        address executor,
+        TokenInfo storage info,
+        address sender,
+        ParaID destinationChain,
+        MultiAddress calldata destinationAddress,
+        uint128 destinationChainFee,
+        uint128 amount
+    ) external returns (Ticket memory ticket) {
+        // Polkadot-native token: burn wrapped token
+        _burnToken(executor, agent, info.token, sender, amount);
+
+        if (destinationChainFee == 0) {
+            revert InvalidDestinationFee();
+        }
```

Comment:
Maybe the `revert InvalidDestinationFee();` will revert the burnToken changes, but just curious to why you don't do the `destinationChainFee` check before the burn?

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/c2adf8f219d90407de1bafcad1ebaab5cf9617fb